### PR TITLE
Remove Microsoft.DotNet.PlatformAbstractions

### LINKED ISSUE #3470: Remove or neuter Microsoft.DotNet.PlatformAbstractions
As agreed more broadly (with @dleeapho et al) we are committed to remove (in some sense) Microsoft.DotNet.PlatformAbstractions from core-setup in 3.0. 

After https://github.com/dotnet/corefx/issues/31002 is completed, all its functionality will have equivalents in CoreFX proper - albeit in different shapes.

(Note that Microsoft.DotNet.DependencyModel is not included here - it must run on older runtimes, so it remains in core-setup.)

Options for M.DN.PA:

1. Remove it, and document alternative functionality. Because the existing package remains on NuGet, we avoid breaking existing code. (However it is possible it becomes out of date, eg., if we change how we determine the RID)
1. Retain it as is, but replace its implementation by calls into the CoreFX functionality, if semantically possible. 
1. As above, but also move into CoreFX.

@eerhardt @dseefeld is the above correct? thoughts?

### PR DESCRIPTION
**Pull request by eerhardt**
This library overlapped with other System APIs and is now obsolete. For API replacement:
```
ApplicationEnvironment.ApplicationBasePath => AppContext.BaseDirectory
HashCodeCombiner => System.HashCode
RuntimeEnvironment.GetRuntimeIdentifier() => RuntimeInformation.RuntimeIdentifier
RuntimeEnvironment.OperatingSystemPlatform => RuntimeInformation.IsOSPlatform(OSPlatform)
RuntimeEnvironment.RuntimeArchitecture => RuntimeInformation.ProcessArchitecture
RuntimeEnvironment.OperatingSystem => RuntimeInformation.OSDescription
RuntimeEnvironment.OperatingSystemVersion => RuntimeInformation.OSDescription / Environment.OSVersion.Version
```
Fix #3470

See also https://github.com/dotnet/docs/issues/18438

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]
I like PRs like this:
image
:smile:

**Comment by AraHaan** [comment]
True.

**Comment by eerhardt** [comment]
FYI - @terrajobst @gewarren @PriyaPurkayastha for the breaking change nature of this.

File: src/installer/managed/CommonManaged.props [hunk 1]
```diff
@@ -34,20 +25,11 @@
...
   </PropertyGroup>
 
-  <PropertyGroup Condition=" '$(TargetFramework)' != 'netstandard2.0' And '$(TargetFramework)' != 'netcoreapp2.0' ">
-    <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
-  </PropertyGroup>
-
   <PropertyGroup>
     <RepositoryType>git</RepositoryType>
-    <RepositoryUrl>git://github.com/dotnet/core-setup</RepositoryUrl>
+    <RepositoryUrl>git://github.com/dotnet/runtime</RepositoryUrl>
```
Comment (ViktorHofer) [inline src/installer/managed/CommonManaged.props:30]: Please remove that property. It's already defined here: https://github.com/dotnet/runtime/blob/b6ef0a54ad0dbe016ec63850e3a70bfacb9bde09/Directory.Build.props#L67.

Comment (ViktorHofer) [reply src/installer/managed/CommonManaged.props:30]: I'm fine if we do that in a follow-up PR.

**Comment by PriyaPurkayastha** [comment]
cc @marklio

**Review by ViktorHofer — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Comment by eerhardt** [comment]
Since this is removing a library, I'll wait until tomorrow before merging this to give it more time for someone to raise concerns.

**Review by dagood — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Comment by AraHaan** [comment]
Why not just use the library to wrap the calls then just ``Obsolete`` with error set to ``true`` that they need to replace calls with what you said above. That way it lets them know that PlatformAbstractions
library is going away soon.

**Comment by eerhardt** [comment]
> Why not just use the library to wrap the calls

There are a few reasons why not to choose this route.

1. The above API mappings aren't able to be wrap around the underlying APIs.
    - See the `Note on OperatingSystem and OperatingSystemVersion` section in 
https://github.com/dotnet/docs/issues/18438.
2. Current users of this library aren't actually "broken" by this change, even though we are treating it like a breaking change. They can continue using the `3.1` version of this library, even in a `net5.0` application. If they try to update the version to `5.0`, they will see that there isn't a new version to update to.
3. That would involve shipping another version of the library, which one of the main goals of this work is to reduce the maintenance cost of this library. Shipping another version adds to the cost.

**Pull request merged**